### ISSUE TITLE
Associate correct compiler metadata for the containerized apps

### ISSUE THREAD

**Issue by v-thakkar** [parca-dev/parca-agent#1366]
Sometimes we associate the 'not so accurate' compiler metadata for the containerized applications, especially when no information regarding the compiler is available. Below is the output of `jps` showing the java applications running on the system.
```
vaishali@pop-os:~$ sudo jps
298407 spring-two-0.0.2-SNAPSHOT.jar
298719 Jps
246237 spring-0.0.1-SNAPSHOT.jar

```
Screenshots of what parca-agent shows for both applications:
1. For the process running on the host
![Screenshot from 2023-02-20 07-09-53](https://user-images.githubusercontent.com/7105009/220023659-7076f19d-21fd-4377-8143-b5db656a78a5.png)

2. For the application running in a docker container(check cgroup_name)
![Screenshot from 2023-02-20 07-06-29](https://user-images.githubusercontent.com/7105009/220023901-1244fff8-a1ba-4c60-87fd-4fed48107920.png)

In the case of Java, it'll be fixed with the ongoing work as part of #1306. But this might be happening for other processes. Opening this so that I don't forget to check and fix it later.

**Comment by v-thakkar** [comment]
Discussed this with @brancz . One idea is to not call the label compiler to avoid confusion for users as this can also be the case for the applications that are statically linking the runtimes of other languages. 

In the case of Java, we'll go for having a separate label that says Java=yes.

**Comment by brancz** [comment]
I would suggest using `java=true`, but I think that's a formality (it's just what's widely used in the Prometheus ecosystem and we want things to feel as familiar as possible). Overall agreed. 💯

**Comment by kakkoyun** [comment]
Fixed by #2580
